01-data-structures/01-introduction-to-data-structures/screen/screen.rb

```diff
@@ -6,18 +6,37 @@ class Screen
   attr_accessor :matrix
 
   def initialize(width, height)
+    self.matrix = []
+    self.width = width
+    self.height = height
+    (0...height).each do |y|
+      row = []
+      (0...width).each do |x|
+        row.push nil
+      end 
+      self.matrix.push(row)
+    end
   end
 
   # Insert a Pixel at x, y
   def insert(pixel, x, y)
+    if inbounds(x, y)     
+      row = self.matrix[y]
+      row[x] = pixel
```

Comment:
Ruby allows for a single line syntax here: `self.matrix[y][x] = pixel if inbounds(x, y)`